module/TueFind/src/TueFind/Controller/MyResearchController.php

```
@@ -72,35 +72,168 @@ public function publishAction()
             return $this->forceLogin();
         }
 
+        $uploadInfos = [];
+        $uploadError = 0;
+
         $dspace = $this->serviceLocator->get(\TueFind\Service\DSpace::class);
         $dspace->login();
         $config = $this->getConfig('tuefind');
 
         $existingRecord = null;
         $dublinCore = null;
+        $controlNumber = null;
         $existingRecordId = $this->params()->fromRoute('record_id', null);
```

The existing record is mandatory, please throw an exception if there is no record id or the record loader fails